Tests/Realm.Tests/Database/AsymmetricObjectTests.cs

```diff
@@ -0,0 +1,382 @@
+        }
+
+        [Test]
+        public void AddHugeAsymmetricObj()
+        {
+            const int ObjectSize = 1_000_000;
+
+            SyncTestHelpers.RunBaasTestAsync(async () =>
+            {
+                var objId = ObjectId.Empty;
+
+                var flxConfig = await GetFLXIntegrationConfigAsync();
+                flxConfig.Schema = new[] { typeof(HugeSyncAsymmetricObject) };
+                using var realm = await GetFLXIntegrationRealmAsync(flxConfig: flxConfig);
+
+                realm.Write(() =>
+                {
+                    var hugeObj = new HugeSyncAsymmetricObject(ObjectSize);
+                    objId = hugeObj.Id;
+                    realm.Add(hugeObj);
+                });
+
+                await realm.SyncSession.WaitForUploadAsync();
+                var documents = await GetObjFromRemoteThroughMongoClient<HugeSyncAsymmetricObject>(flxConfig, "_id", objId);
+                Assert.That(documents.Single().Data.Count, Is.EqualTo(ObjectSize));
+            });
+        }
+
+        [Test]
+        public void AccessAsymmetricObjAfterAddedToRealm_Throws()
+        {
+            SyncTestHelpers.RunBaasTestAsync(async () =>
+            {
+                var partitionLike = Guid.NewGuid().ToString();
+                var flxConfig = await GetFLXIntegrationConfigAsync();
+                flxConfig.Schema = new[] { typeof(BasicAsymmetricObject) };
+                using var realm = await GetFLXIntegrationRealmAsync(flxConfig: flxConfig);
+
+                var asymmetribObj = new BasicAsymmetricObject
+                {
+                    PartitionLike = partitionLike
+                };
+
+                realm.Write(() =>
+                {
+                    realm.Add(asymmetribObj);
+                });
+
+                Assert.Throws<RealmInvalidObjectException>(() =>
+                {
+                    _ = asymmetribObj.PartitionLike;
+                }, "Attempted to access a detached row");
+            });
+        }
+
+        [Test]
+        public void AddSameAsymmetricObjTwice_DoesntThrow()
+        {
+            SyncTestHelpers.RunBaasTestAsync(async () =>
+            {
+                var flxConfig = await GetFLXIntegrationConfigAsync();
+                flxConfig.Schema = new[] { typeof(BasicAsymmetricObject) };
+                using var realm = await GetFLXIntegrationRealmAsync(flxConfig: flxConfig);
+                var partitionLike = Guid.NewGuid().ToString();
+                var asymmetricObj = new BasicAsymmetricObject
+                {
+                    PartitionLike = partitionLike
+                };
+
+                Assert.DoesNotThrow(() =>
+                {
+                    realm.Write(() =>
+                    {
+                        realm.Add(asymmetricObj);
+                    });
+
+                    realm.Write(() =>
+                    {
+                        realm.Add(asymmetricObj);
+                    });
+                });
```

Comment:
Why doesn't this throw? After the first write, we detach the object, then try to add it again - this should be an error as we're trying to add a deleted object to the Realm.